Quadriga/src/main/java/edu/asu/spring/quadriga/web/config/AdminAuthenticationProvider.java

```diff
@@ -0,0 +1,83 @@
+package edu.asu.spring.quadriga.web.config;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.security.authentication.AuthenticationProvider;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.core.GrantedAuthority;
+
+import edu.asu.spring.quadriga.web.login.QuadrigaGrantedAuthority;
+import edu.asu.spring.quadriga.web.login.RoleNames;
+
+/**
+ * 
+ * This class authenticates the Admin user and assigns the appropriate authorities.
+ * @author Chiraag Subramanian 
+ *
+ */
+public class AdminAuthenticationProvider implements AuthenticationProvider {
```

Comment:
where is that used? Why is that needed? Admins should be handled by the regular user service.